Assets/Scripts/Algorithms/Patrolling/CognitiveCoordinated/CognitiveCoordinatedBase.cs

```diff
@@ -34,25 +42,21 @@ protected override void GetDebugInfo(StringBuilder stringBuilder)
         protected override IComponent[] CreateComponents(IRobotController controller, PatrollingMap patrollingMap)
         {
             _controller = controller;
-            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, Coordinator.GlobalMap);
+            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, _globalMap);
             _collisionRecoveryComponent = new CollisionRecoveryComponent(controller, _goToNextVertexComponent);
-
-            return new IComponent[] { _goToNextVertexComponent, _collisionRecoveryComponent };
+            _virtualStigmergyComponent = new VirtualStigmergyComponent<int, Dictionary<int, Vertex>>((_, localKnowledge, _) => localKnowledge, controller);
```

Comment:
Why does the base have a virtual stigmergy component?

Reply:
Done